pkg/config/reload.go

```diff
@@ -148,21 +151,29 @@ func (c *Config) ReloadPauseImage(newConfig *Config) error {
 // ReloadPinnedImages updates the PinnedImages with the provided `newConfig`.
 // The method print log in case of any updates.
 func (c *Config) ReloadPinnedImages(newConfig *Config) {
-	updatedPinnedImages := make([]string, len(newConfig.PinnedImages))
-	updatedPinnedImageList := false
+	if len(newConfig.PinnedImages) == 0 {
+		c.PinnedImages = []string{}
+		return
+	}
+	sort.Strings(newConfig.PinnedImages)
+	sort.Strings(c.PinnedImages)
 
-	for i, image := range newConfig.PinnedImages {
-		if i < len(c.PinnedImages) && image == c.PinnedImages[i] {
+	if cmp.Equal(c.PinnedImages, newConfig.PinnedImages,
+		cmpopts.SortSlices(func(a, b string) bool {
+			return a < b
+		}),
+	) {
+		return
+	}
+	newList := []string{}
+	for _, img := range newConfig.PinnedImages {
+		if img == "" {
 			continue
 		}
-		updatedPinnedImages[i] = image
-		updatedPinnedImageList = true
-	}
-
-	if updatedPinnedImageList {
-		logConfig("pinned_images", strings.Join(updatedPinnedImages, ", "))
-		c.PinnedImages = updatedPinnedImages
+		newList = append(newList, img)
 	}
+	logConfig("pinned_images", strings.Join(newList, ", "))
```

Comment:
We probably don't care about the space?

```suggestion
	logConfig("pinned_images", strings.Join(newList, ","))
```